src/eduid_userdb/credentials/fido.py

```diff
@@ -153,17 +164,24 @@ def __init__(
                 created_ts=created_ts,
             )
 
-        FidoCredential.__init__(self, data)
+        FidoCredential.__init__(self, data, called_directly=called_directly)
+
         self.version = data.pop('version')
         self.public_key = data.pop('public_key')
         self.attest_cert = data.pop('attest_cert', '')
 
-        leftovers = data.keys()
-        if leftovers:
-            if raise_on_unknown:
-                raise UserHasUnknownData('U2F {!r} unknown data: {!r}'.format(self.key, leftovers,))
-            # Just keep everything that is left as-is
-            self._data.update(data)
+        if raise_on_unknown:
+            self.check_unknown_data(data)
+
+        # Just keep everything that is left as-is
+        self._data.update(data)
+
+    @classmethod
+    def from_dict(cls: Type['U2F'], data: Dict[str, Any], raise_on_unknown: bool = True) -> 'U2F':
```

Comment:
Is it possible to use annotations so we can skip the string type of the return value?

`from __future__ import annotations`

Reply:
:+1: